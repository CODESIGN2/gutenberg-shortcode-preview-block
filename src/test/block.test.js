import React from 'react'
import TestUtils from 'react-dom/test-utils';
import ReactDOM from 'react-dom';
import fetchMock from 'fetch-mock';
import MyBlock, { Shortcode } from '../block';
import ShortcodePreview from '../preview';

global.wpApiSettings = {
    root: "http://www.example.com/api/"
};

describe('Shortcode', () => {
    let block, wrapper, wrapperElement;

    beforeEach(() => {
        block = <MyBlock attributes={ {text: "[code]hello world[/code]"} } />;
        wrapper = TestUtils.renderIntoDocument(
            block
        );
        wrapperElement = () => ReactDOM.findDOMNode( wrapper );

		fetchMock.restore();
		fetchMock.mock('*', {html:'<pre>hello world</pre>', js: '', style: ''});
    });

    describe('when added', () => {
        it('Should match Snapshot', () => {
            expect(wrapperElement()).toMatchSnapshot();
        });
    });

    describe('when added in preview mode before receiving a server response', () => {
        it('Should match Snapshot', () => {
            const block = TestUtils.findRenderedComponentWithType(wrapper, Shortcode);
            block.setState( {
				preview: true,
            } );
            expect(wrapperElement()).toMatchSnapshot();
        });
    });

    describe('when added in preview mode with a response', () => {
        it('Should match Snapshot', () => {
            const block = TestUtils.findRenderedComponentWithType(wrapper, Shortcode);
            block.setState( {
				preview: true,
            } );
            const previewComponent = TestUtils.findRenderedComponentWithType(wrapper, ShortcodePreview);
            previewComponent.setState({response: {data: {html: '<pre>hello world</pre>', js: '', style:''}}})
            expect(wrapperElement()).toMatchSnapshot();
        });
    });
});
