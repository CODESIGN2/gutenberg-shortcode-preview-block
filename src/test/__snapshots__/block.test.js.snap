// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shortcode when added Should match Snapshot 1`] = `
<div
  class="wp-block-custom-shortcode components-placeholder"
>
  <label
    class="components-placeholder__label"
    for="blocks-custom-shortcode-input-0"
  >
    <svg
      aria-hidden="true"
      class="dashicon dashicons-editor-code"
      focusable="false"
      height="20"
      role="img"
      viewBox="0 0 20 20"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 6l-4 4 4 4-1 2-6-6 6-6zm2 8l4-4-4-4 1-2 6 6-6 6z"
      />
    </svg>
    Shortcode
  </label>
  <textarea
    class="block-editor-plain-text input-control"
    id="blocks-custom-shortcode-input-0"
    placeholder="Write shortcode hereâ€¦"
    rows="1"
  >
    [code]hello world[/code]
  </textarea>
</div>
`;

exports[`Shortcode when added in preview mode before receiving a server response Should match Snapshot 1`] = `
<div
  class="wp-block-custom-shortcode"
>
  <div
    class="wp-block-embed is-loading"
  >
    <span
      class="components-spinner"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`Shortcode when added in preview mode with a response Should match Snapshot 1`] = `
<div
  class="wp-block-custom-shortcode"
>
  <iframe
    class="components-sandbox"
    height="0"
    sandbox="allow-scripts allow-same-origin allow-presentation"
    title="Preview"
    width="0"
  />
  <div
    class="sandbox__preview-overlay"
  />
</div>
`;
